parcels/fieldset.py

```diff
@@ -164,7 +164,8 @@ def add_constant(self, name, value):
         """
         if name in self.constants:
             raise ValueError(f"FieldSet already has a constant with name '{name}'")
-
+        if isinstance(value, np.timedelta64):
+            value = value / np.timedelta64(1, "s")  # Convert to seconds
         self.constants[name] = np.float32(value)
```

Comment:
I think that it would be better for us to error out when provided non-numeric data, than it would be to convert types under the hood (as then users might expect a type but receive another)